### ISSUE TITLE
Tags not being populated in trace in Tempo

### ISSUE THREAD

**Issue by mvkrishna86** [grafana/agent#1139]
Hi,

I am using grafana agent (1.19.0) as a daemonset in AWS EKS cluster. And every pod pushing logs to http://$(HOSTNAME):4317 where grafana agent is running. I configured the scrape_config as per the documentation, but the trace does not have the namespace, pod and container tags. 

Below is my configuration in grafana agent:
```
traces:
  configs:
    - name: default
      receivers:
        jaeger:
          protocols:
            grpc: null
            thrift_binary: null
            thrift_compact: null
            thrift_http: null
        zipkin: null
        otlp:
          protocols:
            grpc:
            http:
      attributes:
        actions:
          - action: upsert
            key: cluster
            value: qa3
          - action: upsert
            key: region
            value: ap-south-1
          - action: delete
            key: thread.id
          - action: delete
            key: thread.name
      remote_write:
        - endpoint: tempo.nslhub.click:4317
          insecure: false
          headers:
            x-scope-orgid: 285672126725
      batch:
        timeout: 5s
        send_batch_size: 1000
      scrape_configs:
        - bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
          job_name: kubernetes-pods
          kubernetes_sd_configs:
            - role: pod
          relabel_configs:
            - action: replace
              source_labels:
                - __meta_kubernetes_namespace
              target_label: namespace
            - action: replace
              source_labels:
                - __meta_kubernetes_pod_name
              target_label: pod
            - action: replace
              source_labels:
                - __meta_kubernetes_pod_container_name
              target_label: container
          tls_config:
            ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
            insecure_skip_verify: false
      automatic_logging:
        roots: true
        backend:
```

![image](https://user-images.githubusercontent.com/15136443/143771342-c161b0f0-05ba-4e65-bd2d-9dc99cc36911.png)
![image](https://user-images.githubusercontent.com/15136443/143771361-6d92d23e-388e-46fe-a870-98c457083047.png)

**Comment by james-callahan** [comment]
Possibly related to https://github.com/grafana/agent/issues/1095?

**Comment by rfratto** [comment]
@mapno do you know if this would be fixed by #1137?

**Comment by mapno** [comment]
Thanks for the poke. Most likely yes. In that PR we improved pod association, so there are more techniques to associate metadata to the pod sending traces.

These are tag labels `ip`, `net.host.ip`, `k8s.pod.ip`, and `hostname`. And `connection` that will inspect the context of the incoming request to find the IP. We expect those to cover most cases, but it could still be not enough.

If you encounter issues while using the new pod association, let me know!
